src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    dateMap['YYYY'] =
+      dateMap['YY'] < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
```

Comment:
When checking for the presence of 'YYYY' or 'YY' in `dateMap` and deciding to convert between them, you should also verify that the corresponding format actually exists in `fromFormat`. Otherwise, you may end up referencing a format that was never in the original date string.